OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyUtils.cs

```diff
@@ -441,7 +441,7 @@ public static void SetupNameWidget(Widget parent, Session.Client c, OrderManager
 
 			var closed = TranslationProvider.GetString(Closed);
 			var open = TranslationProvider.GetString(Open);
-			slot.GetText = () => truncated.Update(c != null ? c.Name : s.Closed ? closed : open);
+			slot.GetText = () => truncated.Update(c != null ? SanitizePlayerName(c) : s.Closed ? closed : open);
```

Comment:
Shouldn't `SanitizePlayerName` be moved into the/a cached transform as well? Otherwise we do the color lookups all the time.

Reply:
I assume this isn't much of a performance issue as this is just lobby code, though we could add current player color as an argument to the cache

Reply:
I can't cache this by client because the player color can be changed in the lobby.